Open Judge System/OJS.Services/Data/SubmissionsForProcessing/SubmissionsForProcessingDataService.cs

```diff
@@ -9,35 +10,39 @@
 
     public class SubmissionsForProcessingDataService : ISubmissionsForProcessingDataService
     {
-        private GenericRepository<SubmissionForProcessing> SubmissionsForProcessing { get; }
+        private readonly GenericRepository<SubmissionForProcessing> submissionsForProcessing;
 
         public SubmissionsForProcessingDataService(GenericRepository<SubmissionForProcessing> submissionsForProcessing)
         {
-            this.SubmissionsForProcessing = submissionsForProcessing;
+            this.submissionsForProcessing = submissionsForProcessing;
         }
 
         public SubmissionForProcessing GetBySubmissionId(int submissionId) =>
-            this.SubmissionsForProcessing.All().FirstOrDefault(s => s.SubmissionId == submissionId);
+            this.submissionsForProcessing.All().FirstOrDefault(s => s.SubmissionId == submissionId);
 
         public void AddOrUpdate(int submissionId)
         {
-            var submissionForProcessing = this.GetBySubmissionId(submissionId);
+            this.AddOrUpdateWithNoSaveChanges(submissionId);
+            this.submissionsForProcessing.SaveChanges();
+        }
 
-            if (submissionForProcessing != null)
+        //used optimization from this article https://msdn.microsoft.com/en-us/data/jj556205 for better performance
```

Comment:
1/ space after `//`; 2/ Comment shall start with Uppercase letter

Reply:
Do we need this comment very much?

Reply:
This comment was requested by @Innos and could prevent usage of `AutoDetectChangesEnabled` by someone else without reading about it.